veps/sig-network/passt-migration-proposal.md

```diff
@@ -0,0 +1,185 @@
+
+#### How passt-repair Works:
+- Executed as a **one-off process**, separately on migration source and migration target.
+- Communicates with passt over a named Linux Domain Socket.
+- Signals availability and waits for passt requests.
+- Upon receiving a request:
+  - Sets or unsets the `TCP_REPAIR` socket option.
+  - Notifies passt upon completion.
+
+#### Integration with virt-handler
+
+-To support this process, virt-handler will be modified to invoke `passt-repair` in a dedicated goroutine. This goroutine will
+run to completion or timeout, ensuring the migration process does not hang indefinitely.
+
+The invocation is conditioned on the VMI having an interface with `passt` network binding and an enabled feature gate:
+
+1. Before migration starts (source node) in [vmUpdateHelperMigrationSource](https://github.com/kubevirt/kubevirt/blob/release-1.5/pkg/virt-handler/vm.go#L2714) to set the TCP_REPAIR socket option.
+2. Before migration finalizes (target node) in [finalizeMigration](https://github.com/kubevirt/kubevirt/blob/release-1.5/pkg/virt-handler/vm.go#L2714) to set and then clear the TCP_REPAIR socket option.
+
+##### Failure Scenarios:
+passt-repair can fail or timeout if:
+- The Unix Domain Socket created by passt isn't found.
+- Communication with passt is delayed or unresponsive.
+
+Either scenario likely results in broken TCP connections for the guest VM, requiring reconnection.
+- virt-handler does not fail migration on passt-repair errors.
+- passt currently does not abort migration upon TCP_REPAIR failure.
+
+**Note**: Success of passt-repair doesn't guarantee seamless migration alone; sticky IPs from the network provider and CNI are essential.
+
+#### Add passt-repair binary to virt-handler image
+The `passt-repair` binary must be added to the virt-handler container image. The simplest solution is to install the passt RPM. A more secure and storage-efficient solution is extracting only the passt-repair binary from the RPM and installing it standalone.
+
+### Implement vhost-user support in passt sidecar
+vhost-user enables high-performance communication between virtual machines and user-space processes, typically offloading VirtIO emulation to dedicated user-space processes (e.g., DPDK or passt). It improves I/O performance through zero-copy mechanisms and reduced context-switching overhead. Additionally, it is required for seamless TCP migration.
+
+vhost-user must be enabled in the DomainXML. Small code changes are required in passt sidecar code. The format is described [here](https://libvirt.org/formatdomain.html#vhost-user-connection-with-passt-backend).
+
+### Passt CNI
+The passt-cni process is called by `Multus` during primary network interface setup/removal. It runs two sysctl commands:
+1. Allow binding to all ports starting from 0.
+2. Set ping group range to virt-launcher's user ID.
+
+The [passt-binding directory](https://github.com/kubevirt/kubevirt/tree/release-1.5/cmd/cniplugins/passt-binding) will include:
```

Comment:
Why is it in the future tense? I thought we already have it.
What is done here is a suggestion to change it.. and I do not really see if there is a suggestion to change it (e.g. the domainxml).

Reply:
I see.
Please open a different section for deployment. In its current form, everything is mixed together, making it hard to separate the topics discussed.